test/libsolidity/util/TestFileParser.cpp

```diff
@@ -199,13 +203,20 @@ pair<string, bool> TestFileParser::parseFunctionSignature()
 	return {signature, !hasName};
 }
 
-u256 TestFileParser::parseFunctionCallValue()
+TestFileParser::FunctionValue TestFileParser::parseFunctionCallValue()
 {
 	try
 	{
-		u256 value{parseDecimalNumber()};
-		expect(Token::Ether);
-		return value;
+		u256 value{ parseDecimalNumber() };
+		auto token = m_scanner.currentToken();
```

Comment:
Please avoid overusing `auto`, it makes the code harder to read IMO. I think it only makes sense to use `auto` if the actual type is too long AND irrelevant at the same time.

Reply:
Fixed